### PR TITLE
refactor ClientQuerySegmentWalker

### PR DESCRIPTION
**Pull request by jisookim0513**
I have been working on using Spark for long-running Druid queries. I was able to run a long-running druid query without timeout by having intermediate-level brokers that process the query for a specific chunk of segments and letting the final broker merge the query results from intermediate brokers. While doing so, I had to write my own query runners where one does stuff from `RetryQueryRunner` to `toolChest.mergeResults` and the other does stuff from `toolChest.mergeResults` to `FinalizeResultsQueryRunner`. @xvrl suggested refactoring `ClientQuerySegmentWalker` this way so I can use the methods from a new class instead of reusing the code from `ClientQuerySegmentWalker`.

### CODE REVIEW

File: server/src/main/java/io/druid/server/FluentQueryRunnerBuilder.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+package io.druid.server;
```
Comment (navis) [inline server/src/main/java/io/druid/server/FluentQueryRunnerBuilder.java]: missing header

**Comment by CHOIJAEHONG1** [comment]
This looks great.

**Comment by xvrl** [comment]
I'm of course 👍 on this one, but since I was involved I'd prefer if we get two other +1 on this one

**Comment by drcrallen** [comment]
:+1:  needs 1 more

**Comment by b-slim** [comment]
LGTM, It will be nice if we add some unit tests to cover the new class.
BTW IT is not a blocking issue.

**Comment by nishantmonu51** [comment]
👍 , LGTM, @jisookim0513: minor nit, could you also modify QueryRunnerTestHelper.makeXXXQueryRunner to use the new api  ?

**Comment by jisookim0513** [comment]
@nishantmonu51 sure, I will modify QueryRunnerTestHelper to use the new API.

**Comment by gianm** [comment]
👍

**Comment by jisookim0513** [comment]
@nishantmonu51: modified QueryRunnerTestHelper and refactored FluentQueryRunnerBuiler not to take query as a constructor argument (also moved it to druid.io.processing from druid.io.server)

**Comment by xvrl** [comment]
👍 on the latest changes.

**Pull request merged**